src/core/core.scale.js

```diff
@@ -926,9 +909,14 @@ class Scale extends Element {
 		const numTicks = me.ticks.length;
 		const tickWidth = 1 / Math.max(numTicks - (offset ? 0 : 1), 1);
 
-		return index < 0 || index > numTicks - 1
-			? null
-			: me.getPixelForDecimal(index * tickWidth + (offset ? tickWidth / 2 : 0));
+		if (index < 0) {
```

Comment:
I think I'd change the `@return` to `{number=}` (or `{number|null}`)

Reply:
That was going to be a real pain because then we need to check for a `null` return value everytime we call it and handle that case appropriately. And in many instances there's no clear way in which `null` should be handled. I think that throwing an exception here would be the best solution because an `index < 0` is clearly invalid. I'm not quite sure why we try to avoid throwing errors throughout the codebase. But this seemed like the second best solution to me

Reply:
Think about category scale that is zoomed (start/end defined). Asking for a pixel for any of the labels is a valid scenario and a undefined or null result is valid result, when the label is not visible.

Reply:
I've updated this a bit. But I don't think it is valid to call `getPixelForTick` with a negative number or a number greater than the number of labels if the category scale is zoomed. If the category scale is zoomed then in `buildTicks` we do the following:
```
// If we are viewing some subset of labels, slice the original array
labels = (min === 0 && max === labels.length - 1) ? labels : labels.slice(min, max + 1);
me._numLabels = labels.length;
```
If you call `getPixelForTick(0)` it will return the pixel for the first visible tick.